x-pack/legacy/plugins/ml/server/routes/annotations.ts

```diff
@@ -0,0 +1,150 @@
+import { annotationServiceProvider } from '../models/annotation_service';
+import { wrapError } from '../client/error_wrapper';
+import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { RouteInitialization } from '../new_platform/plugin';
+import {
+  deleteAnnotationSchema,
+  getAnnotationsSchema,
+  indexAnnotationSchema,
+} from '../new_platform/annotations_schema';
+
+import { ANNOTATION_USER_UNKNOWN } from '../../common/constants/annotations';
+
+function getAnnotationsFeatureUnavailableErrorMessage() {
+  return Boom.badRequest(
+    i18n.translate('xpack.ml.routes.annotations.annotationsFeatureUnavailableErrorMessage', {
+      defaultMessage:
+        'Index and aliases required for the annotations feature have not been' +
+        ' created or are not accessible for the current user.',
+    })
+  );
+}
+/**
+ * Routes for annotations
+ */
+export function annotationRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup Annotations
+   *
+   * @api {post} /api/ml/annotations Gets annotations
+   * @apiName GetAnnotations
+   * @apiDescription Gets annotations.
+   *
+   * @apiParam {String[]} jobIds List of job IDs
+   * @apiParam {String} earliestMs
+   * @apiParam {Number} latestMs
+   * @apiParam {Number} maxAnnotations Max limit of annotations returned
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} annotations
+   */
+  router.post(
+    {
+      path: '/api/ml/annotations',
+      validate: {
+        body: schema.object(getAnnotationsSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { getAnnotations } = annotationServiceProvider(context);
+        const resp = await getAnnotations(request.body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Annotations
+   *
+   * @api {put} /api/ml/annotations/index Index annotation
+   * @apiName IndexAnnotations
+   * @apiDescription Index the annotation.
+   *
+   * @apiParam {Object} annotation
+   * @apiParam {String} username
+   */
+  router.put(
+    {
+      path: '/api/ml/annotations/index',
+      validate: {
+        body: schema.object(indexAnnotationSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const annotationsFeatureAvailable = await isAnnotationsFeatureAvailable(
+          context.ml!.mlClient.callAsCurrentUser
+        );
+        if (annotationsFeatureAvailable === false) {
+          throw getAnnotationsFeatureUnavailableErrorMessage();
+        }
+
+        const { indexAnnotation } = annotationServiceProvider(context);
+        const username = _.get(request, 'auth.credentials.username', ANNOTATION_USER_UNKNOWN);
```

Comment:
There doesn't seem to be an `auth` object as part of the `request` with this NP switch, so this is always getting set to 'unknown'. We must have to get the logged in username some other way.

![image](https://user-images.githubusercontent.com/7405507/74156074-86ea2880-4c0d-11ea-9157-0ae703b2c867.png)

Reply:
Looks like this is no longer available in the NP request object. It has to be accessed via the NP security plugin api - added in a4ad5a5039fd1c809bbb121a9bc327a63ce8a53b